### ISSUE TITLE
Model.init() doesn't work after dropDatabase() when Model is initialized before

### ISSUE THREAD

**Issue by ulrichb** [Automattic/mongoose#6796]
In the following scenario ...
```JS
const EventModel = mongoose.model('Event',
    new Schema({ eventStreamId: Schema.Types.ObjectId, eventStreamVersion: Number, /*..*/ })
        .index({ eventStreamId: 1, eventStreamVersion: 1 }, { unique: true }));

// ...

await mongoose.connect(/*...*/);

await mongoose.connection.dropDatabase();

// ...

await EventModel.init();
```

... it is possible that the indexes are not created after `await EventModel.init()`. Happens every second time on my local machine.

After moving the `const EventModel = mongoose.model(...)` _after_ `dropDatabase()` it works.

=> It seems that although the collection and the indexes are rebuilt after `dropDatabase()`, the `init()` call doesn't wait for the (new indexes).

mongoose: 5.2.6, mongodb: 4.0, node: 10.7

**Comment by vkarpov15** [comment]
By default, Mongoose starts building indexes when you declare the model with `mongoose.model()`. If you want to manually call `init()` to build indexes, use [`mongoose.connect(uri, { autoIndex: false })`](http://mongoosejs.com/docs/api.html#mongoose_Mongoose-connect). Does that work for you?

I'm curious, why are you declaring your models before calling `mongoose.connect()`?

**Comment by rjanas** [comment]
I have the same issue.
Declaring model after calling `mongoose.connect()` doesn't change anything - it still works every second time.

**Comment by ulrichb** [comment]
Hi @vkarpov15 

> I'm curious, why are you declaring your models before calling mongoose.connect()

That's because my models are declared in ES module global consts next to my repositories (which is nice, because they are hidden in the modules, initialized once and accessible in all repo methods). Then I had the problem that the module initializing occurred before the data access initialization (including `connect()` and `dropDatabase()`) => The situation described above.

I fixed this issue by early initializing the data access and later initialize my container (which initializes my models). => The problem is now fixed for me.

**BUT:** I think issue this should still be solved, because I think this is a pitfall, and (especially when the indexes are data access-logic relevant, like unique indexes for concurrency control) and there for a "bug waiting to happen".
